shared/chat/manage-channels/edit-channel-container.js

```diff
@@ -10,26 +10,38 @@ import {anyWaiting} from '../../constants/waiting'
 
 const mapStateToProps = (state: TypedState, {navigateUp, routePath, routeProps}) => {
   const conversationIDKey = routeProps.get('conversationIDKey')
-  const teamname =
-    routeProps.get('teamname') || Constants.getTeamNameFromConvID(state, conversationIDKey) || ''
-  const waitingForSave = anyWaiting(
+  if (!conversationIDKey) {
+    throw new Error('conversationIDKey unexpectedly empty')
+  }
+
+  const teamname = routeProps.get('teamname')
+  if (!teamname) {
+    throw new Error('teamname unexpectedly empty')
+  }
+
+  const channelInfo = Constants.getChannelInfoFromConvID(state, conversationIDKey)
```

Comment:
This data is already in `state.chat2.metaMap` after you've loaded this conversation. Right now `_getChannels` in actions/teams gets all the chat inbox items for the channels and throws away everything that is not relevant to this screen. I agree that having the duplicated data in the store isn't that nice, maybe a better flow is to have `_getChannels` delegate to a chat action to get the trusted metas for the channels. Then the channel names & descriptions can live in the chat `metaMap` and `getChannelInfoFromConvID` can look there for the data. But :sheep: if you don't want to do that here it's fine.

Reply:
I took a look and yeah, maybe having `_getChannels` write its info to the chat `metaMap` would be a better flow. I'll see if I can do that after my follow-up PR after this one.